pkg/varlinkapi/system.go

```diff
@@ -90,22 +88,25 @@ func (i *LibpodAPI) GetInfo(call iopodman.VarlinkCall) error {
 		Graph_status:         graphStatus,
 	}
 
-	if len(info) > 2 {
-		registriesInterface := info[2].Data["registries"]
-		if registriesInterface != nil {
-			registries = registriesInterface.([]string)
-		}
-	}
-	if len(info) > 3 {
-		insecureRegistriesInterface := info[3].Data["registries"]
-		if insecureRegistriesInterface != nil {
-			insecureRegistries = insecureRegistriesInterface.([]string)
+	if len(info) > 1 {
+		for key, val := range info[2].Data {
```

Comment:
This  looks suspicious. We check for `> 1` but access the third item which can lead to invalid memory accesses if `len(info) == 2`.

Can we add comments to explain the assumptions of the length check and accesses?

Reply:
Fixed